Services/communication_service.c

```diff
@@ -37,155 +38,205 @@ SAT_returnState communication_service_app(csp_packet_t *packet) {
   int8_t status;
 
   /* Comment for PR:
-   * 1- Perhaps initial value assignment should occur here if we want to
-   * call getX after setX command.
-   * 2- Also I think I'll merge most of these structs.
-   * 3- Steps 2-5 in each case can be replaced by a single function.
+   * Steps 2-5 in each case can be replaced by a single function.
    */
-  struct temp_utc temp_temp;
-  struct Sband_config S_config;
-  struct Sband_PowerAmplifier S_PA;
-  struct Sband_Encoder S_enc;
-  struct Sband_Firmware S_Firmware;
-  struct Sband_Status S_status;
-  struct Sband_buffer S_buffer;
 
-  switch (ser_subtype) {
-    case GET_TEMP:
-      // Step 1: get the data
-      HAL_comm_getTemp(&temp_temp.sensor_temperature);
-      // Step 2: convert to network order
-      temp_temp.sensor_temperature = csp_hton32(temp_temp.sensor_temperature);
-      // step3: copy data & status byte into packet
-      status = 0;
-      memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t)); // 0 for success
-      memcpy(&packet->data[OUT_DATA_BYTE], &temp_temp.sensor_temperature, sizeof(uint32_t));
-      // Step 4: set packet length
-      set_packet_length(packet, sizeof(int8_t) + sizeof(uint32_t) + 1); // +1 for subservice
-      // Step 5: return packet
-      if (queue_response(packet) != SATR_OK) {
-        return SATR_ERROR;
-      }
+  Sband_config S_config;
+  Sband_PowerAmplifier S_PA;
+  Sband_Encoder S_enc;
+  Sband_Status S_status;
+  Sband_TR S_TR;
+  Sband_Housekeeping S_HK;
+  Sband_Buffer S_buffer;
```

Comment:
I have to get rid of these and use the structs inside S_FS. Probably will do the same with the previous ones once I merge them into a single general subservice as get_full_status.